File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque
def solution(queue1, queue2):
    answer = 0
    
    q1 = deque(queue1)
    q2 = deque(queue2)
    
    sum1 = sum(q1)
    sum2 = sum(q2)
    if (sum1+sum2)%2!=0:
        return -1
    while(sum1!= sum2):
        if answer > 600000:
            return -1
        if sum1 > sum2:
            tmp = q1.popleft()
            q2.append(tmp)
            sum1 -= tmp
            sum2 += tmp
        elif sum2 > sum1:
            tmp = q2.popleft()
            q1.append(tmp)
            sum1 += tmp
            sum2 -= tmp
        else:
            return answer
        answer+=1
    
    return answer

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = 0;
        
        Queue<Integer> q1 = new LinkedList<>();
        Queue<Integer> q2 = new LinkedList<>();
        
        long sum1 = Arrays.stream(queue1).sum();
        long sum2 = Arrays.stream(queue2).sum();
        
        
        for(int i=0; i<queue1.length;i++){
            q1.add(queue1[i]);
            q2.add(queue2[i]);
        }
        
        while(sum1 != sum2){
            if(answer>(queue1.length + queue2.length)*2){
                return -1;
            }
            int val = 0;
            if(sum1<sum2){
                val = q2.poll();
                q1.add(val);
                sum1 += val;
                sum2 -= val;
            }else if(sum1>sum2){
                val = q1.poll();
                q2.add(val);
                sum1 -= val;
                sum2 += val;
            }else{
                return answer;
            }
            answer++;
        }
        
        return answer;
    }
}